content/PHPDataObjects/DataObjects.q.md

```diff
@@ -0,0 +1,1214 @@
+
+- It executes the argument `INSERT INTO students (first_name, last_name, birthdate, gender) VALUES ('John', 'Smith', '1999-02-10', 'Male')")` in the `query()` method.
+
+
+:::
+
+
+:::
+
+/// type=REPL, readonly=true, filename=[connection.php,student.php]
+
+```php
+// connection.php
+<?php
+    $host = 'localhost';
+    $db = 'LibraryDB';
+    $port = '5432';
+    $username = 'postgres';
+    $password = 'Admin01';
+
+    $dsn = "pgsql:host=$host;port=$port;dbname=$db;user=$username;password=$password";
+
+    try {
+        $conn = new PDO($dsn);
+        if ($conn) {
+            echo "Successfully connected to the database. ";
+        }
+    } catch (Exception $e) {
+        echo "Unable to establish a connection."; 
+    }
+?>
+```
+
+```php
+// student.php
+<?php
+    require_once ("connection.php");
+
+    $stmt = $conn->query("SELECT * FROM students");
+    foreach ($stmt as $row) {
+        echo $row['student_id'].$row['first_name'].$row['last_name'].$row['birthdate'].$row['gender']. ' ';
+    }
+?>
+```
+/// type=SS, answer=[5]
+
+Execute the program. What is its output?
+
+- It produces an error.
+
+- No output is displayed.
+
+- It prints `Unable to establish a connection.`.
+
+- It prints `Successfully connected to the database.` and `JohnSmith1999-10-02Male`.
+
+- It prints `Successfully connected to the database.` and `1JohnSmith1999-02-10Male`.
+
+
+/// type=MS, answer=[3,5]
+
+Which statements correctly describe the `foreach` construct on line 5 of `student.php`?
+
+- It removes all the elements in `$stmt`.
+
+- It accesses all the elements in `$stmt`.
+
+- It iterates through each element in the array.
+
+- It removes all the elements in the multidimensional array `$stmt`.
+
+- It assigns the value of the current element on each iteration to `$row`.
+
+
+/// type=SS, answer=[2]
+
+What value is assigned to `student_id`?
+
+- `0`
+
+- `1`
+
+- `John`
+
+- `Smith`
+
+- No value is assigned.
+
+
+/// type=MS, answer=[3,4,5]
+
+Which statements correctly describe `$stmt = $conn->query("SELECT * FROM students");` on line 4 of `student.php`?
+
+- It assigns the value of `$conn` to `$stmt`.
+
+- It ends the execution of the `query()` method. 
+
+- It calls the `query()` method of the `$conn` object.
+
+- It executes the argument `SELECT * FROM students` in the `query()` method.
+
+- It assigns the result set as a `PDOStatement` object to the variable `$stmt`.
+
+
+:::
+
+
++++
+
+
++++
+
+### Part 2: Knowledge Assessment
+
+/// type=MS, answer=[2,5]
+
+Which statements correctly describe `PDO::query()`?
+
+- It sends queries or uploads data to certain databases.
+
+- It executes an SQL statement in a single function call.
+
+- It defines an interface in PHP to access certain databases.
+
+- It represents a connection between a database server and PHP.
+
+- It returns a result set as a `PDOStatement` object or on failure a `false` boolean value.
+
+
+/// type=SS, answer=[2]
+
+What parameter is passed in the `PDO::query()` method?
```

Comment:
Incorrect term use...

Is `parameter` the right term to use here or `argument`? Please check. Thanks.